src/tools/pkcs11-tool.c

```diff
@@ -2577,6 +2602,24 @@ static void decrypt_data(CK_SLOT_ID slot, CK_SESSION_HANDLE session,
 		mech.pParameter = iv;
 		mech.ulParameterLen = iv_size;
 		break;
+	case CKM_AES_GCM:
+		use_aead_buffer = TRUE;
+		iv_size = 16;
+		iv = get_iv(opt_iv, &iv_size);
+		gcm_params.pIv = iv;
+		gcm_params.ulIvLen = iv_size;
+		gcm_params.ulIvBits = iv_size * 8; // no one seems to know what this is for
+
+		aad_size = 0;
+		aad = get_aad(opt_aad_file, &aad_size);
+		gcm_params.pAAD = aad;
+		gcm_params.ulAADLen = aad_size;
+
+		gcm_params.ulTagBits = opt_tag_len * 8;
```

Comment:
I would sanity-check that the value is in the bounds required by the specs. From the above linked section, it should be between 0 and 128